app/routes.py

```diff
@@ -1,2 +1,349 @@
+
+##################################################################################
+######### Wave_2: Custom endpoints for Rental: Many to Many relationship #########
+##################################################################################
+
+#rental checkout 
+@rentals_bp.route("/check-out", methods=["POST"], strict_slashes=False)
+def rental_checkout():
+
+    rental = request.get_json()
+
+# The API should return back detailed errors and a status 400: Bad Request if the video does not have any available inventory before check out
+#check-out of invalid customer ID and video ID
+    if not isinstance(rental["customer_id"], int) or not isinstance(rental["video_id"], int):
+        return {
+            "detail": "Invaid Data. Must be an Integer."
+        }, 400
+
+    rental_list = Rental(customer_id = rental["customer_id"],
+                        video_id = rental["video_id"],
+                        due_date = datetime.now() + timedelta(days=7))# adds 7 days to datetime.now(current_date)
+
+
+# The API should return back detailed errors and a status 404: Not Found if the customer does not exist
+    customer = Customer.query.get(rental_list.customer_id)
+    if customer is None:
+        return "Not Found", 404
+
+# # The API should return back detailed errors and a status 404: Not Found if the video does not exist
+    video = Video.query.get(rental_list.video_id)
+    if video is None:
+        return "Not Found", 404
+
+    if video.available_inventory <= 0:
+        return {
+            "message": "Invalid data."
+        }, 400
+
+    customer.checkout_count += 1
+    video.available_inventory -= 1
+
+
+    db.session.add(rental_list)
+    db.session.commit()
+
+    return jsonify({
+        "customer_id": rental_list.customer_id,
+        "video_id": rental_list.video_id,
+        "due_date": rental_list.due_date, #7 days from checked out date
+        "videos_checked_out_count": customer.checkout_count,
+        "available_inventory": video.available_inventory 
+    }), 200
+
+
+@rentals_bp.route("/check-in", methods=["POST"], strict_slashes=False)
+def rental_checkin():
+    request_body = request.get_json()
+
+    if "customer_id" not in request_body or "video_id" not in request_body:
+        return jsonify({
+                "detail": "Invaid Data."
+                }), 404
+
+    if not isinstance(request_body["customer_id"], int) or not isinstance(request_body["video_id"], int):
+        return {
+            "detail": "Invaid Data. Must be an Integer."
+        }, 404
+    
+    customer_id = request_body.get("customer_id")
+    video_id = request_body.get("video_id")
+    
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+    for rental in customer.video:
+        if rental.video_id == video_id:
+            customer.checkout_count -= 1
+            video.available_inventory += 1
+            db.session.delete(rental)
+            db.session.commit()
```

Comment:
This is interesting that there's a loop here. It looks like the idea is that if the same customer has multiple copies of the same video checked out, they are expected to return all the duplicate copies at once. Is that right?